src/go/rpk/pkg/cli/cloud/login.go

```diff
@@ -65,153 +65,90 @@ If none of these are provided, rpk will use the SSO method to login.
 rpk.yaml file unless the --save flag is passed. The cloud authorization 
 token and client ID is always synced.
-
-PROFILE SELECTION
-
-This command by default attempts to populate a new profile that talks to a
-cloud cluster for you. If you have an existing cloud profile, this will select
-it, prompting which to use if you have many. If you have no cloud profile, this
-command will prompt you to select one that exists in your organization. If you
-want to disable automatic profile creation and selection, use --no-profile.
 `,
 		Run: func(cmd *cobra.Command, _ []string) {
 			cfg, err := p.Load(fs)
-			out.MaybeDie(err, "unable to load config: %v", err)
+			out.MaybeDie(err, "rpk unable to load config: %v", err)
 			y, err := cfg.ActualRpkYamlOrEmpty()
-			out.MaybeDie(err, "unable to load config: %v", err)
-			auth := y.Auth(y.CurrentCloudAuth)
-			var cc bool
-			if auth != nil {
-				cc = auth.HasClientCredentials()
-			}
-			_, err = oauth.LoadFlow(cmd.Context(), fs, cfg, auth0.NewClient(cfg.DevOverrides()), noBrowser)
+			out.MaybeDie(err, "rpk unable to load config: %v", err)
+
+			p := y.Profile(y.CurrentProfile)
+			authAct, authVir, clearedProfile, isNewAuth, err := oauth.LoadFlow(cmd.Context(), fs, cfg, auth0.NewClient(cfg.DevOverrides()), noBrowser, forceReload, cfg.DevOverrides().CloudAPIURL)
 			if err != nil {
 				fmt.Printf("Unable to login to Redpanda Cloud (%v).\n", err)
-				if e := (*oauth.BadClientTokenError)(nil); errors.As(err, &e) && cc {
+				if e := (*oauth.BadClientTokenError)(nil); errors.As(err, &e) && authVir != nil && authVir.HasClientCredentials() {
 					fmt.Println(`You may need to clear your client ID and secret with 'rpk cloud logout --clear-credentials',
 and then re-specify the client credentials next time you log in.`)
 				} else {
 					fmt.Println(`You may need to clear your credentials with 'rpk cloud logout --clear-credentials', and login again`)
 				}
 				os.Exit(1)
 			}
-
-			if cc && save {
-				yAct, _ := cfg.ActualRpkYaml() // must exist due to LoadFlow checking
-				yAct.Auth(yAct.CurrentCloudAuth).ClientSecret = auth.ClientSecret
-				err = yAct.Write(fs)
+			if authVir.HasClientCredentials() && save {
+				authAct.ClientSecret = authVir.ClientSecret
+				err = y.Write(fs)
 				out.MaybeDie(err, "unable to save client ID and client secret: %v", err)
 			}
-			fmt.Println("Successfully logged in.")
-			if noProfile {
+
+			fmt.Printf("Successfully logged into organization %q (%s) via %s.\n", authAct.Organization, authAct.OrgID, authAct.AnyKind())
+			if !isNewAuth {
+				fmt.Println("If you are trying to authenticate to a different organization, use 'rpk cloud login --reload'.")
+			}
+			fmt.Println()
+
+			// When you have no profile, or you have one that is not selected.
+			if p == nil || len(y.Profiles) == 0 {
+				fmt.Println(`To create an rpk profile to talk to an existing cloud cluster, use 'rpk cloud cluster use'.
```

Comment:
See the changes we discussed in person (change to the language we designed).

Reply:
This now only prints if you use --no-profile -- new messaging is better, and I think this is worth keeping for the --no-profile flag